pkg/api/meta/restmapper.go

```diff
@@ -206,23 +206,32 @@ func (m *DefaultRESTMapper) ResourcesFor(resource unversioned.GroupVersionResour
 
 	case hasGroup:
 		requestedGroupResource := resource.GroupResource()
-		for currResource := range m.pluralToSingular {
-			if currResource.GroupResource() == requestedGroupResource {
-				ret = append(ret, currResource)
+		for plural, singular := range m.pluralToSingular {
+			if singular.GroupResource() == requestedGroupResource {
+				ret = append(ret, plural)
+			}
+			if plural.GroupResource() == requestedGroupResource {
+				ret = append(ret, plural)
 			}
 		}
 
 	case hasVersion:
-		for currResource := range m.pluralToSingular {
-			if currResource.Version == resource.Version && currResource.Resource == resource.Resource {
-				ret = append(ret, currResource)
+		for plural, singular := range m.pluralToSingular {
```

Comment:
Yeah, I gathered, hence the LGTM. :)

On Thu, Feb 11, 2016 at 10:55 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/meta/restmapper.go
> <https://github.com/kubernetes/kubernetes/pull/20968#discussion_r52648403>
> :
>
> >  			}
> >  		}
> >
> >  	case hasVersion:
> > -		for currResource := range m.pluralToSingular {
> > -			if currResource.Version == resource.Version && currResource.Resource == resource.Resource {
> > -				ret = append(ret, currResource)
> > +		for plural, singular := range m.pluralToSingular {
>
> I am skeptical that this is a useful case. If I know I'm looking for a
> "v1" "Pod" and my choices are the main api or the metrics group, one of
> those is super wrong.
>
> This is a problem today. If I ask for "v1, pods", I get an answer. If I
> ask for "v1, pod" I get a failure. This change simply makes them both
> return the same answer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20968/files#r52648403>.
>